flink-tests/src/test/java/org/apache/flink/test/scheduling/ManualRescalingITCase.java

```diff
@@ -0,0 +1,199 @@
+import org.apache.flink.runtime.testtasks.BlockingNoOpInvokable;
+import org.apache.flink.runtime.testutils.CommonTestUtils;
+import org.apache.flink.runtime.testutils.MiniClusterResourceConfiguration;
+import org.apache.flink.test.util.MiniClusterWithClientResource;
+import org.apache.flink.util.TestLogger;
+
+import org.junit.ClassRule;
+import org.junit.Test;
+
+import java.util.function.Supplier;
+
+/** Tests for the manual rescaling of Flink jobs using the REST API. */
+public class ManualRescalingITCase extends TestLogger {
+
+    private static final int NUMBER_OF_SLOTS = 4;
+
+    @ClassRule
+    public static MiniClusterWithClientResource miniClusterWithClientResource =
+            new MiniClusterWithClientResource(
+                    new MiniClusterResourceConfiguration.Builder()
+                            .setConfiguration(createConfiguration())
+                            .setNumberSlotsPerTaskManager(NUMBER_OF_SLOTS)
+                            .build());
+
+    private static Configuration createConfiguration() {
+        final Configuration configuration = new Configuration();
+
+        configuration.set(JobManagerOptions.SCHEDULER, JobManagerOptions.SchedulerType.Adaptive);
+        configuration.set(WebOptions.REFRESH_INTERVAL, 50L);
+        configuration.set(HeartbeatManagerOptions.HEARTBEAT_INTERVAL, 50L);
+
+        return configuration;
+    }
+
+    @Test
+    public void testManualUpScalingWithNewSlotAllocation() throws Exception {
+        final JobVertex jobVertex = new JobVertex("Single operator");
+        final int initialParallelism = 1;
+        final int parallelismAfterRescaling = 2;
+        jobVertex.setParallelism(initialParallelism);
+        jobVertex.setInvokableClass(BlockingNoOpInvokable.class);
+        final JobGraph jobGraph = JobGraphTestUtils.streamingJobGraph(jobVertex);
+
+        runRescalingTest(
+                jobGraph,
+                () ->
+                        JobResourceRequirements.newBuilder()
+                                .setParallelismForJobVertex(jobVertex.getID(), 1, 2)
+                                .build(),
+                initialParallelism,
+                parallelismAfterRescaling,
+                NUMBER_OF_SLOTS - 2);
+    }
+
+    @Test
+    public void testManualUpScalingWithNoNewSlotAllocation() throws Exception {
+        final int initialRunningTasks = 3;
+        final int runningTasksAfterRescale = 4;
+
+        final JobVertex jobVertex1 = new JobVertex("Operator1");
+        jobVertex1.setParallelism(1);
+        jobVertex1.setInvokableClass(BlockingNoOpInvokable.class);
+        final JobVertex jobVertex2 = new JobVertex("Operator2");
+        jobVertex2.setParallelism(2);
+        jobVertex2.setInvokableClass(BlockingNoOpInvokable.class);
+
+        final SlotSharingGroup slotSharingGroup = new SlotSharingGroup();
+        jobVertex1.setSlotSharingGroup(slotSharingGroup);
+        jobVertex2.setSlotSharingGroup(slotSharingGroup);
```

Comment:
Does this work with different slot sharing groups?

Reply:
Sure, I've added an extra IT test to cover that. Take a look at `#testManualUpScalingWithDifferentSlotSharingGroups`.